packages/drivers/routerlicious-driver/src/policies.ts

```diff
@@ -54,4 +54,10 @@ export interface IRouterliciousDriverPolicies {
 	 * Default: true
 	 */
 	enableLongPollingDowngrade: boolean;
+	/**
+	 * Indicates that the container is ephemeral.
+	 * Artifacts relates to the container are limited to container lifetime.
+	 * Default: false
+	 */
+	isEphemeralContainer?: boolean;
```

Comment:
I think there should be no optional fields in here (enableDiscovery included), given that RouterliciousDocumentServiceFactory takes Partial<IRouterliciousDriverPolicies>.

isEphemeralContainer ideally does not belongs here. Given this is per-container property, and factory can create many containers, it feels like it should not be here, but rather an argument to IDocumentServiceFactory.createContainer(). That said, it's problematic to do so, as that impacts not just R11s driver.

If we got with factory approach, we need to somehow clearly document across layers that this option will apply to every container created by such factory instance.

Q: I assume we do not need to pass that flag for all APIs, i.e. loading container is fine as is? That would be ideal, but worth asking.

Reply:
Would changing the phrasing here to useEphemeralContainers fromisEphemeralContainer feel better?

Reply:
Removed the optional fields and tested with cases when the new flags are sent and when they aren't to make sure that nothing is breaking.

I explored moving this as an argument to createContainer but it is making the plumbing from Azure client and webpack quite complicated. I had a detailed conversation with Nicholas about this and we thought it might be an overkill to make the changes in the many layers across container, loader etc to allow 1 tenant to create ephemeral and non-ephemeral containers with 1 AzureClient/DocumentServiceFactory instance. Also like Zach mentioned to me, we already have a strong precedent for passing all "behavior altering" r11s-specific flags into the Driver via the DocumentServiceFactory constructor. So, I am planning to check this in for now. Please do reach out to me if you think that the reasoning is flawed.

> Q: I assume we do not need to pass that flag for all APIs, i.e. loading container is fine as is? That would be ideal, but worth asking.

That is correct. We are passing this only during the container creation.